### PR TITLE
Add opt to use AZ for APISeverLoadBalancer

### PR DESCRIPTION
**Pull request by nguyenhuukhoi**
**What this PR does / why we need it**: We need this option to use in case of having edge 

This PR allows to choose AZ for API Load Balancer. It will be useful if we want to use edge sites,

/ok-to-test

### CODE REVIEW

File: api/v1alpha7/zz_generated.conversion.go [hunk 1]
```diff
@@ -470,14 +470,10 @@ func autoConvert_v1beta1_APIServerLoadBalancer_To_v1alpha7_APIServerLoadBalancer
...
 	}
+	// WARNING: in.AvailabilityZone requires manual conversion: does not exist in peer-type
 	return nil
 }
 
-// Convert_v1beta1_APIServerLoadBalancer_To_v1alpha7_APIServerLoadBalancer is an autogenerated conversion function.
-func Convert_v1beta1_APIServerLoadBalancer_To_v1alpha7_APIServerLoadBalancer(in *v1beta1.APIServerLoadBalancer, out *APIServerLoadBalancer, s conversion.Scope) error {
-	return autoConvert_v1beta1_APIServerLoadBalancer_To_v1alpha7_APIServerLoadBalancer(in, out, s)
-}
-
```
Comment (huxcrux) [inline api/v1alpha7/zz_generated.conversion.go:480]: Same here we probably do not want to remove this function since it is in use

**Review by huxcrux — COMMENTED** [review]
The removal of functions in the conversion seems to break CI. We probably want to fix some kind of migration even if it is loosy?

**Comment by dulek** [comment]
I took the freedom to try to fix this patch, please check out: https://github.com/nguyenhuukhoi/cluster-api-provider-openstack/pull/1

**Comment by nguyenhuukhoi** [comment]
Hello. Thank you for your help. I am busy on this week. I will continue
next week.

**Comment by nguyenhuukhoi** [comment]
> I took the freedom to try to fix this patch, please check out: [nguyenhuukhoi#1](https://github.com/nguyenhuukhoi/cluster-api-provider-openstack/pull/1)

Hello. I do a mistake. Could you help me revert it?

**Comment by dulek** [comment]
I submitted a new PR where I rebased this: #2004.

**Pull request closed without merging**